dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+
+        boolean isLocked = false;
+
+        if (null != requiresCheckout) {
+
+            isLocked = (requiresCheckout instanceof Boolean)?
+                        Boolean.class.cast(requiresCheckout):
+                        DbConnectionFactory.isDBTrue(requiresCheckout.toString());
+        }
+
+        return isLocked;
+    }
+
+    private String createIntermediateTableForeignKeyActionId() {
+
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEY_ACTION_ID;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEY_ACTION_ID;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEY_ACTION_ID;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEY_ACTION_ID;
+        }
+
+        return H2_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEY_ACTION_ID;
+    }
+
+    private String createIntermediateTableForeignKeyStepId() {
+
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID;
+        }
+
+        return H2_CREATE_INTERMEDIATE_TABLE_FOREIGN_KEYS_STEP_ID;
+    }
+
+    /**
+     * Verifies if the unique index in the {@code workflow_scheme_x_structure} table exists.
+     * @return
+     */
+    private String selectTableIndex() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_SELECT_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_SELECT_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_SELECT_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_SELECT_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Creates the {@code workflow_scheme_x_structure} without the unique index.
+     * @return
+     */
+    private String createTableIndex() {
+        if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_CREATE_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_CREATE_WORKFLOW_SCHEME_X_STRUCTURE_INDEX;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Drops the {@code workflow_scheme_x_structure} unique index.
+     * @return
+     */
+    private String dropTableIndex() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)